tests/autodiff/was/warped-sampling-1d.slang

```diff
@@ -0,0 +1,296 @@
+    float k = 1.0 / (((x_dist * x_dist) + no_diff(boundaryTerm(aux_isect))));
+    return k;
+}
+
+[BackwardDifferentiable]
+float attachToGeometry(Intersection isect)
+{
+    float leftWt = detach(isect.ls.x1 - isect.x);
+    float rightWt = detach(isect.x - isect.ls.x0);
+
+    return (leftWt * isect.ls.x0 + rightWt * isect.ls.x1) / (leftWt + rightWt);
+}
+
+[BackwardDifferentiable]
+float warp(Intersection isect, inout PRNG prng)
+{
+    float totalWeight = 0.f;
+    float totalWarpedPoint = 0.f;
+
+    float aux_sigma = 0.01;
+
+    for (int i = 0; i < 32; i++)
+    {
+        float y = no_diff(sample1DNormal(prng, isect.x, aux_sigma));
+        float y_flipped = 2 * isect.x - y;
+
+        Intersection aux_isect_left = intersect(y);
+
+        if (aux_isect_left.isIntersected)
+        {
+            float pdf = pdf1DNormal(y, isect.x, aux_sigma);
+            float wt = harmonicWeight(isect, aux_isect_left) * (pdf / detach(pdf));
+            totalWarpedPoint += attachToGeometry(aux_isect_left) * wt;
+            totalWeight += wt;
+        }
+        
+        Intersection aux_isect_right = intersect(detach(y_flipped));
+
+        if (aux_isect_right.isIntersected)
+        {
+            float pdf = pdf1DNormal(y_flipped, isect.x, aux_sigma);
+            float wt = harmonicWeight(isect, aux_isect_right) * (pdf / detach(pdf));
+            totalWarpedPoint += attachToGeometry(aux_isect_right) * wt;
+            totalWeight += wt;
+        }
+    }
+
+    return totalWarpedPoint / totalWeight;
+}
+
+[BackwardDifferentiable]
+Intersection warpedIntersect(float x, inout PRNG prng)
+{   
+    // TODO: For now the jacobian here is 1.0, 
+    // but we will need to adjust the warp by the jacobian for
+    // more complex intersection models.
+    //
+    Intersection isect = intersect(x);
+
+    Intersection.Differential d_isect = Intersection.Differential.dzero();
+    d_isect.x = 1.0;
+
+    var dpwarp = infinitesimal(
+        __fwd_diff(warp)(diffPair(isect, d_isect), prng));
+
+    isect.x = detach(isect.x) + dpwarp.p;
+    isect.wt = isect.wt * (1 + dpwarp.d);
+
+    return isect;
+}
+
+[BackwardDifferentiable]
+float renderSample(inout PRNG prng) 
+{
+    float u = no_diff(prng.nextFloat1D());
+    
+    float leftBound = 0.0;
+    float rightBound = 1.0;
+
+    float sample = leftBound * u + rightBound * (1 - u);
+    float weight = 1.0/(rightBound - leftBound);
+ 
+    Intersection isect = warpedIntersect(sample, prng);
+
+    return shadeIntersection(isect) * isect.wt;
+}
+
+[numthreads(1000, 1, 1)]
+void computeMain(uint3 threadIdx : SV_DispatchThreadID,)
+{
+    uint seed = (threadIdx.x * threadIdx.x) * 30 + 3;
+    PRNG prng = PRNG(seed);
+    
+    float d_color = 1.0 / 1000.0;
+    __bwd_diff(renderSample)(prng, d_color);
+
+    AllMemoryBarrierWithGroupSync();
+
+    if (threadIdx.x < 10)
+        endpointDifferentialBuffer[threadIdx.x] =
+            reinterpret<float, uint>(endpointDifferentialBufferUInt[threadIdx.x]);
```

Comment:
Don't do this, floating point results are not determinstic between builds and execution environments.

Reply:
I'm not sure how else to do `atomicAdd` on floats? Should I compare the uint value directly?